Sources/Commands/Utilities/TestingSupport.swift

```diff
@@ -102,6 +102,15 @@ enum TestingSupport {
     ) throws -> [TestSuite] {
         // Run the correct tool.
         #if os(macOS)
+        let targetTriple = try swiftTool.getTargetToolchain().targetTriple
+        guard targetTriple.darwinPlatform == .macOS else {
```

Comment:
@MaxDesiatov thought I'd point out that this is a good opportunity to use your `matches` method from #6819 — in fact the check is currently not strong enough because `swiftpm-xctest-helper` can't handle x64 tests on arm64 macOS without explicitly requesting Rosetta translation (not to mention that AS-on-Intel is also not trivially possible). What do you think about renaming `matches` to something along the lines of `isRuntimeCompatible(with:)` since those are the emergent semantics?

Reply:
Also, I wonder whether it's worth a spike to support iOS simulator testing on macOS — though iiuc we'd need a simulator `swiftpm-xctest-helper` binary for that.

Reply:
> What do you think about renaming `matches` to something along the lines of `isRuntimeCompatible(with:)` since those are the emergent semantics?

I like that idea, thanks! I've updated the corresponding PR 👍

Reply:
> though iiuc we'd need a simulator swiftpm-xctest-helper binary for that.

Probably, yah. Note that `swiftpm-xctest-helper` isn't part of SwiftPM itself anymore for a bit, so you would need to file a FB for that.